### PR TITLE
chore: set up release-please

### LINKED ISSUE #563: Set up release-please
We are moving to release-please for most of our repos, so we need to get it set up here.

Additional non-code setup:

- [x] Update [organization Actions secrets][1] to give this repository access to `NPM_TOKEN`, `MASTODON_ACCESS_TOKEN`, and the four `TWITTER_*` secrets.
- [x] Update [repository Actions settings][2] to "Allow GitHub Actions to create and approve pull requests".
- [ ] Update [npm package settings][3] to allow publishing with 2FA **or** automation or granular access tokens.

[1]: https://github.com/organizations/eslint/settings/secrets/actions
[2]: https://github.com/eslint/espree/settings/actions
[3]: https://www.npmjs.com/package/@eslint/espree/access

See https://github.com/eslint/create-config/pull/54 as an example.

### PR DESCRIPTION
**Pull request by btmills**
Referencing the checklist from #563, I was still able to complete the first two independently, but one of the TSC members will need to complete the third item:

- [x] Update [npm package settings](https://www.npmjs.com/package/@eslint/espree/access) to allow publishing with 2FA **or** automation or granular access tokens.

In setting this up previously on other repositories, some packages already had this setting enabled, so don't be surprised if that's the case.

Fixes #563

To ease review, this differs from https://github.com/eslint/create-config/blob/9fc2de4eb76dd04218b5a4c72a6006563c74f589/.github/workflows/release-please.yml only by the package name:

```diff
diff --git ../create-config/.github/workflows/release-please.yml ./.github/workflows/release-please.yml
index 49defac..371943e 100644
--- ../create-config/.github/workflows/release-please.yml
+++ ./.github/workflows/release-please.yml
@@ -14,7 +14,7 @@ jobs:
         id: release
         with:
           release-type: node
-          package-name: '@eslint/create-config'
+          package-name: 'espree'
           pull-request-title-pattern: 'chore: release${component} ${version}'
       - uses: actions/checkout@v3
         if: ${{ steps.release.outputs.release_created }}
@@ -27,14 +27,14 @@ jobs:
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
         if: ${{ steps.release.outputs.release_created }}
-      - run: 'npx @humanwhocodes/tweet "@eslint/create-config ${{ steps.release.outputs.tag_name }} has been released: ${{ steps.release.outputs.html_url }}"'
+      - run: 'npx @humanwhocodes/tweet "espree ${{ steps.release.outputs.tag_name }} has been released: ${{ steps.release.outputs.html_url }}"'
         if: ${{ steps.release.outputs.release_created }}
         env:
           TWITTER_CONSUMER_KEY: ${{ secrets.TWITTER_CONSUMER_KEY }}
           TWITTER_CONSUMER_SECRET: ${{ secrets.TWITTER_CONSUMER_SECRET }}
           TWITTER_ACCESS_TOKEN_KEY: ${{ secrets.TWITTER_ACCESS_TOKEN_KEY }}
           TWITTER_ACCESS_TOKEN_SECRET: ${{ secrets.TWITTER_ACCESS_TOKEN_SECRET }}
-      - run: 'npx @humanwhocodes/toot "@eslint/create-config ${{ steps.release.outputs.tag_name }} has been released: ${{ steps.release.outputs.html_url }}"'
+      - run: 'npx @humanwhocodes/toot "espree ${{ steps.release.outputs.tag_name }} has been released: ${{ steps.release.outputs.html_url }}"'
         if: ${{ steps.release.outputs.release_created }}
         env:
           MASTODON_ACCESS_TOKEN: ${{ secrets.MASTODON_ACCESS_TOKEN }}

```

### CODE REVIEW

**Review by nzakas — APPROVED** [review]
LGTM. I've updated the `espree` package's npm settings to allow for CI tokens to publish.

Just waiting for @mdjermanovic as he is determining when to roll out release-please to more packages.

**Review by snitin315 — APPROVED** [review]
LGTM

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          release-type: node
+          package-name: 'espree'
+          pull-request-title-pattern: 'chore: release${component} ${version}'
```
[the remark this answers is not in the document]
Comment (btmills) [reply .github/workflows/release-please.yml:18]: Looks like the CLA checker is unhappy with the web suggestion commit. Let me know if you'd like me to redo that commit locally and force push.

[the remark this answers is not in the document]
Comment (mdjermanovic) [reply .github/workflows/release-please.yml:18]: I think it's fine to merge this, that's a confirmed bug in the checker: https://github.com/communitybridge/easycla/issues/3903

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**